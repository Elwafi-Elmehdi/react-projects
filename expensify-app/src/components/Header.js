import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
	<div>
		<h1>Expensify</h1>
		<NavLink activeClassName="active" to="/" exact={true}>
			Dashboard
		</NavLink>
		<NavLink activeClassName="active" to="/create">
			Create Expense
		</NavLink>
	</div>
);

export default Header;
