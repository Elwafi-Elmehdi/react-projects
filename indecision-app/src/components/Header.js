import React from "react";
const Header = (props) => (
	<div className="header">
		<h1 className="header__title">{props.title}</h1>
		<h2 className="header__sub-title">{props.subTitle}</h2>
	</div>
);

Header.defaultProps = {
	title: "some default value",
};
export default Header;
