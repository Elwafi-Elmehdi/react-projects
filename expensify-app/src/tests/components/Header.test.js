// React Test Renderer npm module used for rendering cmps in files js
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Header from "../../components/Header";

test("Should render header component", () => {
	const renderer = new ReactShallowRenderer();
	renderer.render(<Header />);
	console.log(renderer.getRenderOutput());
});
