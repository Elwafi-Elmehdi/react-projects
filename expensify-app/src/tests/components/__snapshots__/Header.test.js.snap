// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header component 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
</div>
`;
